pkg/daemon/ceph/osd/daemon.go

```diff
@@ -202,34 +269,111 @@ func addDirsToDirMap(dirList []string, dirMap *map[string]int) {
 	}
 }
 
-func loadOSDDirMap(kv *k8sutil.ConfigMapKVStore, nodeName string) (map[string]int, error) {
-	dirMapRaw, err := kv.GetValue(getConfigStoreName(nodeName), osdDirsKeyName)
-	if err != nil {
-		return nil, err
+func getRemovedDevices(agent *OsdAgent) (*config.PerfScheme, *DeviceOsdMapping, error) {
+	removedDevicesScheme := config.NewPerfScheme()
+	removedDevicesMapping := &DeviceOsdMapping{Entries: map[string]*DeviceOsdIDEntry{}}
+
+	if !oposd.IsRemovingNode(agent.devices) {
+		// TODO: support more removed device scenarios beyond just entire node removal
+		return removedDevicesScheme, removedDevicesMapping, nil
 	}
 
-	var dirMap map[string]int
-	err = json.Unmarshal([]byte(dirMapRaw), &dirMap)
+	scheme, err := config.LoadScheme(agent.kv, config.GetConfigStoreName(agent.nodeName))
 	if err != nil {
-		return nil, err
+		return nil, nil, fmt.Errorf("failed to load agent's partition scheme: %+v", err)
 	}
 
-	return dirMap, nil
+	for _, entry := range scheme.Entries {
+		if !entry.IsCollocated() {
+			// TODO: support removal of distributed OSD partitions
+			logger.Warningf("distributed OSD partitions are not currently supported for removal: %+v", entry)
```

Comment:
what is it about distributed partitions that we can't support removal? seems like we need to support it